acvm/src/lib.rs

```diff
@@ -159,8 +159,8 @@ pub trait ProofSystemCompiler {
     /// if the language and proof system does not line up.
     fn np_language(&self) -> Language;
 
-    // Returns true if the backend supports the selected black box function
-    fn black_box_function_supported(&self, opcode: &BlackBoxFunc) -> bool;
+    // Returns true if the backend supports the selected opcode
+    fn opcode_supported(&self, opcode: &Opcode) -> bool;
```

Comment:
We can create a stubbed `ProofSystemCompiler` here in a test similar how we do for `PartialWitnessGenerator`.

It doesn't match by itself but we can wrap it in a closure.

Reply:
The function trait is `Fn` trait so the &self should not matter here -- I can quickly add an example backend in a test to show for the PR

Reply:
Added a test with a dummy compiler that calls the compile method -- its very verbose though, so I'd not want to keep it since its testing compile time behaviour.

Reply:
Didn't see @TomAFrench reply until I refreshed the page :)

Reply:
@kevaundray No worries. A lot of my experiments with pluggable backends also remove `self` so now I have an extra reason to remove them in the future 😛